src/CMakeLists.txt

```diff
@@ -3,4 +3,90 @@
 # Copyright 2023 LunarG, Inc.
 #
 # SPDX-License-Identifier: Apache-2.0
+
+add_library(VulkanCompilerConfiguration INTERFACE)
+add_library(Vulkan::CompilerConfiguration ALIAS VulkanCompilerConfiguration)
+if(${CMAKE_CXX_COMPILER_ID} MATCHES "(GNU|Clang)")
+    target_compile_options(VulkanCompilerConfiguration INTERFACE
+        -Wpedantic
+        -Wunreachable-code
+        -Wunused-function
+        -Wall
+        -Wextra
+        -Wpointer-arith
+    )
+    if(${CMAKE_CXX_COMPILER_ID} MATCHES "Clang")
+        target_compile_options(VulkanCompilerConfiguration INTERFACE
+            -Wunreachable-code-return
+            -Wconversion
+            -Wimplicit-fallthrough
+            -Wstring-conversion
+        )
+    endif()
+elseif(MSVC)
+    target_compile_options(VulkanCompilerConfiguration INTERFACE
+        /W4
+        /we5038 # Enable warning about MIL ordering in constructors
+    )
+
+    # Enforce stricter ISO C++
+    target_compile_options(VulkanCompilerConfiguration INTERFACE /permissive-)
+
+    # Speed up Visual Studio builds
+    if (MSVC_IDE)
+        target_compile_options(VulkanCompilerConfiguration INTERFACE /MP)
+    endif()
+
+    # Minimize what Windows.h leaks
+    target_compile_definitions(VulkanCompilerConfiguration INTERFACE NOMINMAX WIN32_LEAN_AND_MEAN)
+endif()
+
+target_compile_definitions(VulkanCompilerConfiguration INTERFACE VK_ENABLE_BETA_EXTENSIONS)
+if(WIN32)
+    target_compile_definitions(VulkanCompilerConfiguration INTERFACE VK_USE_PLATFORM_WIN32_KHR)
+elseif(ANDROID)
+    target_compile_definitions(VulkanCompilerConfiguration INTERFACE VK_USE_PLATFORM_ANDROID_KHR)
+elseif(APPLE)
+    target_compile_definitions(VulkanCompilerConfiguration INTERFACE VK_USE_PLATFORM_METAL_EXT)
+    if (IOS)
+        target_compile_definitions(VulkanCompilerConfiguration INTERFACE VK_USE_PLATFORM_IOS_MVK)
+    else()
+        target_compile_definitions(VulkanCompilerConfiguration INTERFACE VK_USE_PLATFORM_MACOS_MVK)
+    endif()
```

Comment:
I'm not actually sure this is correct, as I believe VK_USE_PLATFORM_MACOS_MVK and VK_USE_PLATFORM_IOS_MVK are deprecated. 

https://registry.khronos.org/vulkan/specs/1.3-extensions/html/chap55.html#VK_MVK_ios_surface

Reply:
Ah. I didn't know. Cool beans.

Reply:
Actually even though it's deprecated VVL still has code for both of these compile definitions. So I'd rather keep it.